gauge-browser-agent-otel/tests/step_implementation.js

```diff
@@ -0,0 +1,107 @@
+/* globals gauge*/
+"use strict";
+const path = require("path");
+const {
+  openBrowser,
+  write,
+  closeBrowser,
+  screenshot,
+  click,
+  into,
+  textBox,
+  goto,
+  $,
+  evaluate,
+} = require("taiko");
+const assert = require("assert");
+const headless = process.env.headless_chrome.toLowerCase() === "true";
+const config = require("../config");
+const jwtDecode = require("jwt-decode");
+const axios = require("axios");
+
+const decoded = jwtDecode(config.apiToken); // Agent Token
+const baseUrl = decoded["x-sl-server"]; // Base url of the backend
+
+const createTestSession = () => {
+  return axios.post(
+    baseUrl.replace("/api", "/sl-api/v1/test-sessions"), // Public API to start a test session
+    {
+      testStage: "Gauge Tests",
+      bsid: "51721270-ead5-498b-b22f-c3f9861ea44e", // Better set from environment variable or get from window.$Sealights
+    },
+    {
+      headers: {
+        Authorization: `Bearer ${config.apiToken}`,
+      },
+    }
+  );
+};
+
+// Works the same with any test framework, before test starts emit event to set baggage
+beforeScenario(async (scenario) => {
+  // Start a test session
+  const { testSessionId } = (await createTestSession()).data.data;
```

Comment:
We need to split this in the demo code according to [Gauge's life-cycle](https://github.com/getgauge/gauge-js/blob/master/docs/syntax/execution-hooks.md) and add support to all the of the TIA public API life-cycle:
1.  Creating a test-session should be done in the beforeSuite.
2. We need to add a call to the DELETE `/sl-api/v1/test-sessions/:testSessionId` at the afterSuite.
3. We need to add a call to the POST `/sl-api/v1/test-sessions/:testSessionId` in the beforeScenario for the test-start
4. We need to add a call to the POST `/sl-api/v1/test-sessions/:testSessionId` in the afterScenario for the test-end

Reply:
For reporting test events I guess the API is changed to v2?
`POST /sl-api/v2/test-sessions/{testSessionId}` ?

https://sealights.atlassian.net/wiki/spaces/SUP/pages/2690220039/Test+Sessions+API+a.k.a+TIA+API#Sending-Test-Events

Reply:
Also do we need to call this API twice? Or just once in the afterScenario like so:

`POST /sl-api/v2/test-sessions/{testSessionId}
Authorization: Bearer {sealights-agent-token}
Content-Type: application/json
[
  {
    "name": "TestToRun1",
    "start": 1619341754996,
    "end": 1619341754998,
    "status": "passed"
  }
]`

Reply:
1. Yes, I've updated my comment to v2.
2. You can call it once, but calling it twice will allow us catching an onboarding issue easier.